includes/islandora_ordered_zip_newspaper_batch.inc

```diff
@@ -0,0 +1,228 @@
+    $page_number = 1;
+    foreach ($this->objectInfo as $sequence => $info) {
+      if ($sequence != 'MODS') {
+        $children[] = new IslandoraOrderedNewspaperPageBatchObject($connection, $this->id, $page_number, $info, $this->preprocessorParameters);
+        $page_number += 1;
+      }
+    }
+    return $children;
+  }
+
+  /**
+   * Function batchProcess.
+   */
+  public function batchProcess() {
+    module_load_include('inc', 'islandora_newspaper', 'includes/utilities');
+    // This melds together the IslandoraNewspaperIssueBatchObject and
+    // IslandoraNewspaperFlatBatchObject's processes into one as the structure
+    // of the preprocess has changed.
+    $newspaper = islandora_object_load($this->parentNewspaper);
+    $issues = islandora_newspaper_get_issues($newspaper);
+    $num_issues = count($issues) + 1;
+    $this->relationships->add(ISLANDORA_RELS_EXT_URI, 'isSequenceNumber', (string) $num_issues, RELS_TYPE_PLAIN_LITERAL);
+    $this->label = $this->getTitle();
+    $this->getMods();
+    $this->getDc();
+    return ISLANDORA_BATCH_STATE__DONE;
+  }
+
+  /**
+   * Function getMods.
+   */
+  protected function getMods() {
+    if (!isset($this['MODS'])) {
+      $mods_datastream = $this->constructDatastream('MODS', 'M');
+      $mods_datastream->mimetype = 'application/xml';
+      $mods_datastream->label = 'MODS Record';
+      if (isset($this->objectInfo['MODS'])) {
+        $mods_datastream->setContentFromFile($this->objectInfo['MODS']->uri, FALSE);
+      }
+      if (!isset($mods_datastream->content)) {
+        // If we do not yet have any XML, create a dumb MODS with just the
+        // title, and set it...
+        $title = $this->baseName;
+
+        $mods_datastream->content = <<<EOXML
+<mods xmlns:mods="http://www.loc.gov/mods/v3" xmlns="http://www.loc.gov/mods/v3">
+  <titleInfo>
+    <title>$title</title>
+  </titleInfo>
+</mods>
+EOXML;
+      }
+      $this->ingestDatastream($mods_datastream);
+    }
+    return $this['MODS']->content;
+  }
+
+  /**
+   * Function batchProcess.
+   */
+  protected function getDc() {
+    if (!isset($this['DC'])) {
+      // Create the DC datastream on this object.
+      $dc_datastream = $this->constructDatastream('DC', 'X');
+      $dc_datastream->mimetype = 'application/xml';
+      $dc_datastream->label = 'DC Record';
+
+      // Get the DC by transforming from MODS.
+      if (!isset($dc_datastream->content)) {
+        // XXX: Might want to make this use a file, instead of staging the DC
+        // in the database table (inside the object we serialize).
+        $new_dc = static::runXslTransform(array(
+          'xsl' => drupal_get_path('module', 'islandora_batch') . '/transforms/mods_to_dc.xsl',
+          'input' => $this->getMods(),
+        ));
+        if ($new_dc) {
+          $dc_datastream->content = $new_dc;
+        }
+      }
+
+      if (isset($dc_datastream->content)) {
+        $this->ingestDatastream($dc_datastream);
+      }
```

Comment:
I know [this is copied](https://github.com/Islandora/islandora_book_batch/blob/7.x/includes/islandora_book_batch.inc#L326-L328), but... Doesn't really make sense: We don't add the datastream to the object, so the [later return](https://github.com/discoverygarden/islandora_ordered_zip_newspaper_batch/pull/1/files#diff-61d558dab8488b8d736ffbde2c317cdcR204) would error... Or return `NULL`? Almost thinking we should explicitly throw an exception or something, when we don't ingest...

... Shrug?